### PR TITLE
Bump semver

### PR DESCRIPTION
**Pull request by henrymercer**
Bumps semver to fixed versions (see https://github.com/advisories/GHSA-c2qf-rxjj-qqgw).  Use overrides to address dependencies on semver that come from packages that haven't been updated yet.  This is a precaution as the attack vector here (ReDoS) is likely not relevant to the Action.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,6 +75,18 @@
...
+    },
+    "@actions/tool-cache": {
+      "semver": ">=6.3.1"
+    },
+    "eslint-plugin-import": {
+      "semver": ">=6.3.1"
+    },
+    "eslint-plugin-jsx-a11y": {
+      "semver": ">=6.3.1"
+    },
```
Comment (aeisenberg) [inline package.json:89]: Can you explain why this isn't sufficient? 

```suggestion
    "semver": "^7.5.4",
```

Comment (henrymercer) [reply package.json:89]: It's not possible to specify an override for a direct dependency.  Furthermore, I'd prefer to force upgrade dependencies to the same major version, to reduce the chances of breaking something.

Comment (henrymercer) [reply package.json:89]: I'd prefer to force upgrade dependencies to the same major version, to reduce the chances of breaking something.

Comment (aeisenberg) [reply package.json:89]: thanks.

**Review by aeisenberg — APPROVED** [review]
Deleting 16k lines is not bad either.

**Pull request merged**